worker/handler.ts

```diff
@@ -78,6 +80,111 @@ const ga: Reporter = async (...args) => {
   await Promise.all([gaForBots(...args), gaForHumans(...args)]);
 };
 
+function ga4(
+  request: Request,
+  conn: ConnInfo,
+  response: Response,
+  _start: number,
+  error?: unknown,
+) {
+  Promise.resolve().then(async () => {
+    // We're tracking page views and file downloads. These are the only two
+    // HTTP methods that _might_ be used.
+    if (!/^(GET|POST)$/.test(request.method)) {
+      return;
+    }
+
+    const bot = isBot(request);
+    const botName = bot ? getBotName(request) : null;
+
+    // If the visitor is using a web browser, only create events when we serve
+    // a top level documents or download; skip assets like css, images, fonts.
+    if (!bot && !isNonAssetRequest(request, response) && error == null) {
+      return;
+    }
```

Comment:
Here too. It seems less error prone if we are non relying on these hardcoded heuristics.